plugins/outputs/bigquery/bigquery.go

```diff
@@ -0,0 +1,215 @@
+package bigquery
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"reflect"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/influxdata/telegraf"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/option"
+)
+
+const (
+	testingHostEnv     = "BIGQUERY_TESTING_HOST"
+	defaultOffSetKey   = "offset-key.json"
+	timeStampFieldName = "timestamp"
+)
+
+type BigQuery struct {
+	CredentialsFile string `toml:"credentials_file"`
+	Project         string `toml:"project"`
+	Dataset         string `toml:"dataset"`
+
+	client *bigquery.Client
+}
+
+var sampleConfig = `	
+  ## Credentials File
+  credentials_file = "/path/to/service/account/key.json"
+  ## GCP Project
+  project = "my-gcp-project"
+
+  ## The namespace for the metric descriptor
+  dataset = "telegraf"
+`
+
+func (b *BigQuery) Connect() error {
+	if b.Project == "" {
+		return fmt.Errorf("Project is a required field for BigQuery output")
+	}
+
+	if b.Dataset == "" {
+		return fmt.Errorf("Dataset is a required field for BigQuery output")
+	}
+
+	b.setUpClient()
+
+	return nil
+}
+
+func (b *BigQuery) setUpClient() error {
+	if endpoint, present := os.LookupEnv(testingHostEnv); present {
+		return b.setUpTestClient(endpoint)
+	}
+
+	return b.setUpDefaultClient()
+}
+
+func (b *BigQuery) setUpTestClient(endpoint string) error {
+	noAuth := option.WithoutAuthentication()
+	endpoints := option.WithEndpoint("http://" + endpoint)
+
+	ctx := context.Background()
+
+	if c, err := bigquery.NewClient(ctx, b.Project, noAuth, endpoints); err != nil {
+		return err
+	} else {
+		b.client = c
+		return nil
+	}
```

Comment:
As you return in the if you can save some indention:
```suggestion
	}
	b.client = c

	return nil
```